From afc69363f5c1c386f445a4604e598e098203acb5 Mon Sep 17 00:00:00 2001
From: Aikar <aikar@aikar.co>
Date: Mon, 31 Dec 2012 01:25:46 -0500
Subject: [PATCH] Anvil Event

Fire an event for anvil usage, remove cap from server, send level to client
---
 .../java/net/minecraft/server/ContainerAnvil.java  | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/src/main/java/net/minecraft/server/ContainerAnvil.java b/src/main/java/net/minecraft/server/ContainerAnvil.java
index 16ec67569..cb79b857e 100644
--- a/src/main/java/net/minecraft/server/ContainerAnvil.java
+++ b/src/main/java/net/minecraft/server/ContainerAnvil.java
@@ -7,6 +7,10 @@ import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
 // CraftBukkit start
+import com.domnian.paperdragon.events.AnvilEvent; // PaperDragon
+import org.bukkit.Bukkit; // PaperDragon
+import org.bukkit.entity.Player; // PaperDragon
+import org.bukkit.craftbukkit.inventory.CraftItemStack; // PaperDragon
 import org.bukkit.craftbukkit.inventory.CraftInventoryView;
 // CraftBukkit end
 
@@ -277,6 +281,16 @@ public class ContainerAnvil extends Container {
             if (this.levelCost >= 40 && !this.m.abilities.canInstantlyBuild) {
                 itemstack1 = ItemStack.a;
             }
+            // PaperDragon start
+            AnvilEvent event = new AnvilEvent((Player) player.player.getBukkitEntity(), CraftItemStack.asBukkitCopy(itemstack), CraftItemStack.asBukkitCopy(itemstack2), CraftItemStack.asBukkitCopy(itemstack1), this.levelCost);
+            Bukkit.getPluginManager().callEvent(event);
+            if (event.isCancelled()) {
+                itemstack1 = ItemStack.NULL_ITEM;
+            } else {
+                itemstack1 = CraftItemStack.asNMSCopy(event.getResult());
+                this.levelCost = event.getCost();
+            }
+            // PaperDragon end
 
             if (!itemstack1.isEmpty()) {
                 int k2 = itemstack1.getRepairCost();
-- 
2.17.0

